### PR TITLE
Improve Swift style guidelines

### PR DESCRIPTION
**Pull request by richtolley**
Tracked by [PT-514](https://novoda.atlassian.net/browse/PT-514)

## This PR adds

Adds some missing recommendation from the Apple API Guide, mostly general points about naming conventions

## Considerations and implementation

Added a few extra examples.

### Test(s) added 
N/A

### Paired with 
@bertadevant

### CODE REVIEW

File: style-guides/swift/README.markdown [hunk 1]
```diff
@@ -59,6 +62,16 @@ In order to have the compiler enforce the decision above and have your builds fa
 
 ## Naming
 
+###Fundamentals
+Strive to follow the fundamental principles outline in the [API Design Guidelines](https://swift.org/documentation/api-design-guidelines/#follow-case-conventions). Pay particular attention to the following conventions:
+
+* Clarity at the point of use is your most important goal
+* Clarity is more important than brevity
+* Use terminology well. Take care to use terms in a way that don't surprise experts or confuse beginners - follow the [Principle of Least Astonishment (POLA)](https://en.wikipedia.org/wiki/Principle_of_least_astonishment). Use precedent for names where 
+* using precedent for names
```
Comment (FabiaS) [inline style-guides/swift/README.markdown]: I am not totally clear about this:
>` * ...). Use precedent for names where `
>` * using precedent for names`

Is there something missing after the `where`? Should there only be one of the two sentences?

File: style-guides/swift/README.markdown [hunk 2]
```diff
@@ -97,6 +110,34 @@ let uRLString: UrlString
...
+```
+
+**Not Preferred**
+```swift
+let burning: Bool 
+``` 
+
+###Methods and Functions
+
+In general, prefer methods and properties over free functions. Always ensure that the intent of the method is clear at the call site. The [API Design Guidelines](https://swift.org/documentation/api-design-guidelines/) set out a number of conventions for different kinds of method
```
Comment (FabiaS) [inline style-guides/swift/README.markdown]: > `for different kinds of method`

method`-s` ?

File: style-guides/swift/README.markdown [hunk 3]
```diff
@@ -97,6 +110,34 @@ let uRLString: UrlString
 let userId: UserId
 ```
 
+Boolean types should read like assertions:
+
+**Preferred**
+```swift
+let isOnFire: Bool
```
Comment (FabiaS) [inline style-guides/swift/README.markdown]: 👍

File: style-guides/swift/README.markdown [hunk 4]
```diff
@@ -97,6 +110,34 @@ let uRLString: UrlString
...
+  * verb methods follow the -ed, -ing rule for the non-mutating version
+  * noun methods follow the formX rule for the mutating version 
+  * naming methods for their side effects
+  * using names based on roles, not types
+  * striving for fluent usage (as far as possible code should read as idiomatic English prose)
+  * giving the same base name to methods that share the same meaning
+  * avoiding overloads on return type
+  * labeling closure and tuple parameters
+  * taking advantage of default parameters 
+  * place default parameters at the end of an argument tlist
```
Comment (FabiaS) [inline style-guides/swift/README.markdown]: nitpicking :) but if we want to keep the same style for all the conventions it should say `placing`

File: style-guides/swift/README.markdown [hunk 5]
```diff
@@ -97,6 +110,34 @@ let uRLString: UrlString
...
+**Not Preferred**
+```swift
+let burning: Bool 
+``` 
+
+###Methods and Functions
+
+In general, prefer methods and properties over free functions. Always ensure that the intent of the method is clear at the call site. The [API Design Guidelines](https://swift.org/documentation/api-design-guidelines/) set out a number of conventions for different kinds of method
+
+  * sometimes compensating for weak type information (e.g the `sender` parameter in an `@IBAction method`)
```
Comment (FabiaS) [inline style-guides/swift/README.markdown]: can we put this line further down? I feel like it is too prominent up here for something that starts with `sometimes` ;) Also can we rephrase it to something like 

> *compensating for weak type information in order to clarify a parameter’s role (e.g. using `button` instead of 'sender' in an `@IBAction method`)

File: style-guides/swift/README.markdown (hunk 4 above)
Comment (richtolley) [reply style-guides/swift/README.markdown]: yes, that's better

File: style-guides/swift/README.markdown (hunk 1 above)
Comment (richtolley) [reply style-guides/swift/README.markdown]: Yes, fixed
